python-sdk/src/astro/sql/operators/merge.py

```diff
@@ -75,6 +75,9 @@ def __init__(
 
     def execute(self, context: Context) -> BaseTable:
         db = create_database(self.target_table.conn_id, table=self.source_table)
```

Comment:
Can we move this line below the if condition, so that we avoid another DB call with that method in case tables do not belong to the same datasource?